File: 백준/Bronze/1085. 직사각형에서 탈출/직사각형에서 탈출.cc
#include<bits/stdc++.h>

using namespace std;

int x,y,w,h,rst1,rst2;

int main() {
	
	cin >> x >> y >> w >> h;
	
	if(x>w-x){
		rst1 += w-x;
	}else rst1 += x;
	if(y>h-y){
		rst2 += h-y;
	}else rst2 += y;
	
	if(rst1>rst2) cout << rst2;
	else cout << rst1;

    return 0;
}

File: 백준/Bronze/1085. 직사각형에서 탈출/직사각형에서 탈출.py
a, b, c, d = map(int,input().split())
rst1 = 0
rst2 = 0

if c-a > a:
    rst1 += a
elif a==c :
    rst1 += 0
else :
    rst1 += c-a

if d-b > b:
    rst2 += b
elif b==d:
    rst2 += 0
else :
    rst2 += d-b


if rst1>rst2:
    print(rst2)
else :
    print(rst1)